sklearn/discriminant_analysis.py

```diff
@@ -586,7 +585,7 @@ def transform(self, X):
                                       "solver (use 'svd' or 'eigen').")
         check_is_fitted(self)
 
-        X = check_array(X)
+        X = self._validate_data(X, dtype=[np.float64, np.float32], reset=False)
```

Comment:
I think the requirement of float64 or float32 in fit is for the linear algebra routines (svd, lsqr, ..). Here in transform, we call `dot` and that should work also if `X` has dtype `int`.

Reply:
If we do not cast here, then we are letting numpy to do the casting during the `np.dot` call or `X - self.xbar_` call. I am okay with this.